cmd/podman/replay_kube.go

```diff
@@ -0,0 +1,247 @@
+		ArgsUsage:              "kubernetes YAML file",
+		UseShortOptionHandling: true,
+		OnUsageError:           usageErrorHandler,
+	}
+)
+
+// generateKubeYAMLCmdgenerates or replays kube
+func replayKubeYAMLCmd(c *cli.Context) error {
+	var (
+		podOptions    []libpod.PodCreateOption
+		podYAML       v1.Pod
+		writer        io.Writer
+		registryCreds *types.DockerAuthConfig
+		namespaces    map[string]string
+		containers    []*libpod.Container
+	)
+
+	ctx := getContext()
+	if rootless.IsRootless() {
+		return errors.Wrapf(libpod.ErrNotImplemented, "rootless users")
+	}
+	args := c.Args()
+	if len(args) > 1 {
+		return errors.New("you can only replay one kubernetes file at a time")
+	}
+	if len(args) < 1 {
+		return errors.New("you must supply at least one file")
+	}
+
+	runtime, err := libpodruntime.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	content, err := ioutil.ReadFile(args[0])
+	if err != nil {
+		return err
+	}
+
+	if err := yaml.Unmarshal(content, &podYAML); err != nil {
+		return errors.Wrapf(err, "unable to read %s as YAML", args[0])
+	}
+
+	podOptions = append(podOptions, libpod.WithInfraContainer())
+	podOptions = append(podOptions, libpod.WithPodName(podYAML.ObjectMeta.Name))
+	// TODO for now we just used the default kernel namespaces; we need to add/subtract this from yaml
+
+	nsOptions, err := shared.GetNamespaceOptions(strings.Split(DefaultKernelNamespaces, ","))
+	if err != nil {
+		return err
+	}
+	podOptions = append(podOptions, nsOptions...)
+	podPorts := getPodPorts(podYAML.Spec.Containers)
+	podOptions = append(podOptions, libpod.WithInfraContainerPorts(podPorts))
+
+	// Create the Pod
+	pod, err := runtime.NewPod(ctx, podOptions...)
+	if err != nil {
+		return err
+	}
+	// Print the Pod's ID
+	fmt.Println(pod.ID())
+
+	podInfraID, err := pod.InfraContainerID()
+	if err != nil {
+		return err
+	}
+
+	namespaces = map[string]string{
+		"pid":  fmt.Sprintf("container:%s", podInfraID),
+		"net":  fmt.Sprintf("container:%s", podInfraID),
+		"user": fmt.Sprintf("container:%s", podInfraID),
+		"ipc":  fmt.Sprintf("container:%s", podInfraID),
+		"uts":  fmt.Sprintf("container:%s", podInfraID),
+	}
+	if !c.Bool("quiet") {
+		writer = os.Stderr
+	}
+
+	dockerRegistryOptions := image2.DockerRegistryOptions{
+		DockerRegistryCreds: registryCreds,
+		DockerCertPath:      c.String("cert-dir"),
+	}
+	if c.IsSet("tls-verify") {
+		dockerRegistryOptions.DockerInsecureSkipTLSVerify = types.NewOptionalBool(!c.BoolT("tls-verify"))
+	}
+
+	for _, container := range podYAML.Spec.Containers {
+		var (
+			containerConfig createconfig.CreateConfig
+			envs            map[string]string
+		)
+
+		containerConfig.Runtime = runtime
+		containerConfig.Image = container.Image
+		newImage, err := runtime.ImageRuntime().New(ctx, container.Image, c.String("signature-policy"), c.String("authfile"), writer, &dockerRegistryOptions, image2.SigningOptions{}, false)
```

Comment:
Can we make this (converting Kube YAML to a container config) a function? I feel like it might be reusable elsewhere

Reply:
yup, done